### DISCUSSION TITLE
Map through array of users and add a record.

### DISCUSSION THREAD

**Discussion by alucardu** [prisma/prisma#9449] [Q&A] [answered]
A User has a field `FollowedBy` that references other Users.

```prisma
model User {
  id                Int     @id @default(autoincrement())
  email             String  @unique
  name              String
  user_name         String  @unique
  password          String
  movies            Movie[]
  notifications     Notification[]
  followedBy        User[]  @relation("UserFollows", references: [id])
  following         User[]  @relation("UserFollows", references: [id])
}
```

I want to create a query that adds a notification in the `Notification` field to each User in the `FollowedBy` field if the current user takes an action.

```prisma
model Notification {
  id                Int     @id @default(autoincrement())
  link              String?
  movie_id          Int?
  message           String
  icon              String?
  thumbnail         String?
  user              User    @relation(fields: [userId], references: [id])
  userId            Int
  watched           Boolean
}
```

I can't find if it's possible to directly add a record to an array field. I thought maybe [SET](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#set) could do it, but I think that's only for setting reference records, not records themself.

So my question is: is it possible to directly add a record to a array field or does Prisma only work through relations in this regard?

**Comment by alucardu** [comment]
I've created this query:

```ts
  createNotification: async (_, args, {req, res}) => {
    const followedBy = await prisma.user.findUnique({
      where: {id: req.userId},
      include: {
        followedBy: true,
      },
    });

    followedBy.followedBy.map(async(user) => {
      await prisma.user.update({
        where: {id: user.id},
        data: {
          notifications: {
            create: {
              message: args.message,
              watched: false,
            },
          },
        },
      });
    });

    return returnUserNotifications(args, req);
  },
  ```

This creates an array of users that follow the current user. Then I map through the results and update each user (who follows the current user) and update their `notification` list. 

This creates a number of notifications:
![image](https://user-images.githubusercontent.com/779921/134825795-200437f0-bdd7-4b48-a5e1-0432dbc7e2df.png)

I kind of feel that having a single table `Notifications` to house every notification made in the entire application will not scale well. But as stated above, I can't see any other way. Also the syntax seems off. It would be better if I could directly update/create the data on the users instead of creating an array first.

**Reply by alucardu** [reply]
I worked on a bit more, found there's an `updateMany` and with the right `where` I get all my users:

```ts
  createNotification: async (_, args, {req, res}) => {
    const followedBy = await prisma.user.updateMany({
      where: {
        following: {
          some: {
            id: req.userId
          },
        },
      },
      data: {
        notifications: {
          create: {
            message: args.message,
            watched: false,
          },
        },
      },
    })
```

But this returns an error:

```
Unknown arg `notifications` in data.notifications for type UserUpdateManyMutationInput. Did you mean `email`? Available args:
type UserUpdateManyMutationInput {
  email?: String | StringFieldUpdateOperationsInput
  name?: String | StringFieldUpdateOperationsInput
  user_name?: String | StringFieldUpdateOperationsInput
  password?: String | StringFieldUpdateOperationsInput
}
```

This is odd to me since the `notifications` list is available on the `User` model (in the topic start). And when I run it like this:

```
      await prisma.user.update({
        where: {id: user.id},
        data: {
          notifications: {
            create: {
              message: args.message,
              watched: false,
            },
          },
        },
      });
```

There's no complain about the `notifications` not being available.

**Comment by ryands17** [comment, marked as answer]
@alucardu 👋 

You would need to go with the first approach where you update records separately using `update` as we don't support updating relation fields in `updateMany` at the moment. It would be great if you could add a 👍 to this feature request so that we can look into this: https://github.com/prisma/prisma/issues/3143

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
